containers/daap-python-base/src/var/task/versioning.py

```diff
@@ -111,6 +113,38 @@ def update_metadata(self, input_data) -> str:
 
         return new_version
 
+    def update_schema(self, input_data: dict, table_name: str):
+        """
+        Create a new version with updated schema.
+        """
+        schema = DataProductSchema(
+            data_product_name=self.data_product_config.name,
+            table_name=table_name,
+            input_data=input_data,
+            logger=self.logger,
+        ).load()
+
+        if not schema.valid:
+            raise InvalidUpdate()
+
+        state, changes = schema_update_type(schema)
+        self.logger.info(f"Update type {state}")
+        self.logger.info(f"Changes to schema: {changes}")
+
+        latest_version = schema.version
+        new_version = generate_next_version_string(schema.version, state)
+        self._copy_from_previous_version(
+            latest_version=latest_version, new_version=new_version
+        )
+        new_version_key = (
+            DataProductConfig(schema.data_product_name)
+            .schema_path(table_name, new_version)
+            .key
+        )
+        schema.convert_schema_to_glue_table_input_csv()
+        schema.write_json_to_s3(new_version_key)
```

Comment:
I think you will also need to write a new version of the metadata.json here, since it has the version in, and the list of schemas

Reply:
Possibly due to having the version within the metadata json, but this isn't currently populated as is an autogenerated field which we haven't done any of yet.

I think schemas will be unchanged because this is for updating and is only existing schemas

Reply:
we should probably make sure the version key is added to metadata and schema on write to s3. Maybe a separate issue though

Reply:
Yeah, maybe that should be a separate ticket then?